include/manager.h

```diff
@@ -0,0 +1,69 @@
+#ifndef _MANAGER_H
+#define _MANAGER_H
+
+#include <assert.h>
+#include <string.h>
+#include <unistd.h>
+#include <dlfcn.h>
+#include <sys/auxv.h>
+
+// vDSO wrapper needed includes
+#include <sys/time.h>
+//#include <sys/vdso.h>
+
+/*******************************************************************************
+ * Compartment
+ ******************************************************************************/
+
+#define MAX_INTERCEPT_COUNT 4
+
+#include "compartment.h"
+
+#define ENV_FIELDS_CNT 1
+extern const char* comp_env_fields[ENV_FIELDS_CNT];
+extern char** environ;
+
+const char* get_env_str(const char*);
+int manager___vdso_clock_gettime(clockid_t, struct timespec*);
+
+struct Compartment* manager_find_compartment_by_addr(void*);
+struct Compartment* manager_find_compartment_by_ddc(void* __capability);
```

Comment:
Isn't this second function the same as the first in the sense that you just get the address of the DDC and pass it to `manager_find_compartment_by_addr`?

Reply:
It could be implemented like that. But in my head, it might make sense to separate the two based on the input parameter and intent... I'm unconvinced though, until I expand the code and use cases for these functions.

Reply:
Up to you: I'd be inclined to start with a single function and only add the second if it later turns out to be useful.

Reply:
I prefer it the other way around: add more functions, then remove them as needed. So I would say for now just leave both in, and adapt appropriately.